### PR TITLE
feat(checkbox, tree-item, card): add component tokens with fallbacks

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #7180

### Checkbox

--calcite-checkbox-background-color: defines the background color of the component
--calcite-checkbox-background-color-checked: defines the background color of the component when it's in a ::checked state
--calcite-checkbox-icon-color: defines the icon color of the checkbox.
--calcite-checkbox-shadow: defines the shadow of the component
--calcite-checkbox-shadow-checked: defines the shadow of the component in a ::checked state
--calcite-checkbox-shadow-hover: defines the shadow of the component in a :hover state.
--calcite-checkbox-shadow-invalid: defines the shadow of the component in an invalid state.
--calcite-checkbox-size: defines the checkbox's size vertically and horizontally.

### Tree Item
--calcite-tree-item-checkbox-background-color: defines the background-color of the sub-component.
--calcite-tree-item-checkbox-background-color-checked: defines the background-color of the sub-component when checked.
--calcite-tree-item-checkbox-icon-color: defines the checkmark color of the sub-component.
--calcite-tree-item-checkbox-shadow: defines the shadow of the sub-component.
--calcite-tree-item-checkbox-shadow-checked: defines the shadow of the sub-component when checked.
--calcite-tree-item-checkbox-shadow-hover: defines the shadow of the sub-component when hovered.
--calcite-tree-item-checkbox-shadow-invalid: defines the shadow of the sub-component when invalid.
--calcite-tree-item-checkbox-size: defines the size of the sub-component.

### Card

--calcite-card-checkbox-background-color: defines the background-color of the sub-component.
--calcite-card-checkbox-background-color-checked: defines the background-color of the sub-component when checked.
--calcite-card-checkbox-icon-color: defines the checkmark color of the sub-component.
--calcite-card-checkbox-shadow: defines the shadow of the sub-component.
--calcite-card-checkbox-shadow-checked: defines the shadow of the sub-component when checked.
--calcite-card-checkbox-shadow-hover: defines the shadow of the sub-component when hovered.
--calcite-card-checkbox-shadow-invalid: defines the shadow of the sub-component when invalid.
--calcite-card-checkbox-size: defines the size of the sub-component.

### CODE REVIEW

File: packages/calcite-components/src/components/card/card.scss [hunk 1]
```diff
@@ -8,6 +8,15 @@
  * @prop --calcite-card-box-shadow: Specifies the box shadow of the component.
  * @prop --calcite-card-corner-radius: Specifies the corner radius of the component.
  *
+ * @prop --calcite-card-checkbox-background-color: defines the background-color of the sub-component.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/card/card.scss:11]: Let's add all of these checkbox ones in as deprecated. We can pick up the new Card tokens for the updated behavior (an internal action with mapped tokens) when `main` gets merged in to `7180`.

File: packages/calcite-components/src/components/card/card.scss [hunk 2]
```diff
@@ -3,18 +3,27 @@
  *
  * These properties can be overridden using the component's tag as selector.
  *
+ *
+ * @prop --calcite-card-accent-color-selected: Specifies the accent color of the component when `selected`.
  * @prop --calcite-card-background-color: Specifies the background color of the component.
  * @prop --calcite-card-border-color: Specifies the border color of the component.
- * @prop --calcite-card-shadow: Specifies the shadow of the component.
+ * @prop --calcite-card-checkbox-background-color-checked: *deprecated* defines the background-color of the sub-component when checked.
+ * @prop --calcite-card-checkbox-background-color: *deprecated* defines the background-color of the sub-component.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/card/card.scss:11]: Maybe a convention outside the scope of this, but can we use **[Deprecated]** for this tag?

File: packages/calcite-components/src/components/card/card.scss [hunk 3]
```diff
@@ -3,18 +3,27 @@
...
+ * @prop --calcite-card-checkbox-icon-color: *deprecated* defines the checkmark color of the sub-component.
+ * @prop --calcite-card-checkbox-shadow-checked: *deprecated* defines the shadow of the sub-component when checked.
+ * @prop --calcite-card-checkbox-shadow-hover: *deprecated* defines the shadow of the sub-component when hovered.
+ * @prop --calcite-card-checkbox-shadow-invalid: *deprecated* defines the shadow of the sub-component when invalid.
+ * @prop --calcite-card-checkbox-shadow: *deprecated* defines the shadow of the sub-component.
+ * @prop --calcite-card-checkbox-size: *deprecated* defines the size of the sub-component.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/card/card.scss:17]: I don't think we need to allow the checkbox size to be customized when it's rendered within another component. Applies to Tree Item as well.

File: packages/calcite-components/src/components/card/card.scss (hunk 2 above)
Comment (driskull) [reply packages/calcite-components/src/components/card/card.scss:11]: Maybe we can have a custom stylelint rule for this? Deprecated should be surrounded by brackets or something.

Comment (macandcheese) [reply packages/calcite-components/src/components/card/card.scss:11]: Yeah, that's a good idea. Both `@prop` here and `@slot` in the `tsx` files use this as unfortunately you can't use the built-in JSDoc `@deprecated` functionality. Maybe something Stencil could build into their compiler.

Comment (driskull) [reply packages/calcite-components/src/components/card/card.scss:11]: We could open a stencil request for this but i'm not sure if they would add it.

File: packages/calcite-components/src/components/card/card.scss (hunk 3 above)
Comment (alisonailea) [reply packages/calcite-components/src/components/card/card.scss:17]: This was a previously documented token though. I think we need to deprecate and remove in the next breaking change

Comment (macandcheese) [reply packages/calcite-components/src/components/card/card.scss:17]: Sorry, I just meant `--calcite-card-checkbox-size` and `--calcite-tree-item-checkbox-size`.

File: packages/calcite-components/src/components/tree-item/tree-item.scss [hunk 4]
```diff
@@ -1,3 +1,18 @@
...
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-tree-item-checkbox-background-color: defines the background-color of the sub-component.
+ * @prop --calcite-tree-item-checkbox-background-color-checked: defines the background-color of the sub-component when checked.
+ * @prop --calcite-tree-item-checkbox-icon-color: defines the checkmark color of the sub-component.
+ * @prop --calcite-tree-item-checkbox-shadow: defines the shadow of the sub-component.
+ * @prop --calcite-tree-item-checkbox-shadow-checked: defines the shadow of the sub-component when checked.
+ * @prop --calcite-tree-item-checkbox-shadow-hover: defines the shadow of the sub-component when hovered.
+ * @prop --calcite-tree-item-checkbox-shadow-invalid: defines the shadow of the sub-component when invalid.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/tree-item/tree-item.scss:12]: I don't think a Checkbox can be in an invalid state within Tree, so this isn't applicable.

File: packages/calcite-components/src/components/card/card.scss [hunk 5]
```diff
@@ -3,18 +3,27 @@
...
+ * @prop --calcite-card-accent-color-selected: Specifies the accent color of the component when `selected`.
  * @prop --calcite-card-background-color: Specifies the background color of the component.
  * @prop --calcite-card-border-color: Specifies the border color of the component.
- * @prop --calcite-card-shadow: Specifies the shadow of the component.
+ * @prop --calcite-card-checkbox-background-color-checked: *deprecated* defines the background-color of the sub-component when checked.
+ * @prop --calcite-card-checkbox-background-color: *deprecated* defines the background-color of the sub-component.
+ * @prop --calcite-card-checkbox-icon-color: *deprecated* defines the checkmark color of the sub-component.
+ * @prop --calcite-card-checkbox-shadow-checked: *deprecated* defines the shadow of the sub-component when checked.
+ * @prop --calcite-card-checkbox-shadow-hover: *deprecated* defines the shadow of the sub-component when hovered.
+ * @prop --calcite-card-checkbox-shadow-invalid: *deprecated* defines the shadow of the sub-component when invalid.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/card/card.scss:15]: This one as well - the Checkbox within Card can't be in an invalid state.

File: packages/calcite-components/src/components/card/card.scss (hunk 2 above)
Comment (driskull) [reply packages/calcite-components/src/components/card/card.scss:11]: Would it be the same as this issue @macandcheese ? https://github.com/ionic-team/stencil/issues/3111

Comment (macandcheese) [reply packages/calcite-components/src/components/card/card.scss:11]: Similar - the JSDoc `@deprecated` they mention in the follow-up comment works for props, events, methods, just not for slots / css props.

File: packages/calcite-components/src/components/card/card.scss [hunk 6]
```diff
@@ -3,18 +3,24 @@
  *
  * These properties can be overridden using the component's tag as selector.
  *
+ *
+ * @prop --calcite-card-accent-color-selected: Specifies the accent color of the component when `selected`.
  * @prop --calcite-card-background-color: Specifies the background color of the component.
  * @prop --calcite-card-border-color: Specifies the border color of the component.
- * @prop --calcite-card-shadow: Specifies the shadow of the component.
+ * @prop --calcite-card-checkbox-background-color-checked: defines the background-color of the sub-component when checked.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/card/card.scss:10]: I’d honestly be ok with not adding these checkbox values at all. The property associated with them on the Card was deprecated with the introduction of the Card Group. It wasn’t possible to theme this explicitly before so we shouldn’t have to worry about maintaining any existing behavior. The Card component has new css props for the equivalent theming need.

**Pull request merged**